system/API/ResponseTrait.php

```diff
@@ -366,7 +372,15 @@ protected function format($data = null)
 
 		// Determine correct response type through content negotiation
 		$config = new Format();
-		$format = $this->request->negotiate('media', $config->supportedResponseFormats, false);
+
+		if (! in_array($this->format, ['json', 'xml']))
```

Comment:
Does hard-coding the available formats prevent meaningful extension? I wonder if this should be checking again the application-configured formatters instead (**app/Config/Format.php**) in case a developer has defined their own.

Reply:
what I understand from `ResourceController` is this feature can set default format for an API. For example, all API by default using json format, but one API will have xml format, so we can overide this by `setFormat` function.

![setFormat](https://user-images.githubusercontent.com/2386359/76583987-4122c780-650d-11ea-94e9-ecc5fd85d5eb.png)

Currently, setFormat only validate two kind of format, either json or xml, that why it need to be re-checked, by hard-coding the available format. 

In case a developer has defined their own format, he can set it up with custom format then system will check again using request negotiate.

@MGatner , tell me if my understanding is wrong.

Reply:
No, you are very right! I didn’t realize that was hard-codes into `setFormat` as well. Might be worth changing these down the road but we’ll leave that for the next pass at the RESTful, maybe with the After Filter.